conjure-java-core/src/integrationInput/java/com/palantir/product/EmptyPathServiceEndpoints.java

```diff
@@ -1,57 +1,80 @@
 import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
 import javax.annotation.Generated;
 
 @Generated("com.palantir.conjure.java.services.UndertowServiceHandlerGenerator")
-public final class EmptyPathServiceEndpoints implements Service {
+public final class EmptyPathServiceEndpoints implements UndertowService {
     private final UndertowEmptyPathService delegate;
 
     private EmptyPathServiceEndpoints(UndertowEmptyPathService delegate) {
         this.delegate = delegate;
     }
 
-    public static Service of(UndertowEmptyPathService delegate) {
+    public static UndertowService of(UndertowEmptyPathService delegate) {
         return new EmptyPathServiceEndpoints(delegate);
     }
 
     @Override
-    public Registrable create(ServiceContext context) {
-        return new EmptyPathServiceRegistrable(context, delegate);
+    public List<Endpoint> endpoints(UndertowRuntime runtime) {
+        return new EmptyPathServiceFactory(runtime, delegate).create();
     }
 
-    private static final class EmptyPathServiceRegistrable implements Registrable {
+    private static final class EmptyPathServiceFactory {
         private final UndertowEmptyPathService delegate;
 
-        private final SerializerRegistry serializers;
+        private final UndertowRuntime runtime;
 
-        private EmptyPathServiceRegistrable(
-                ServiceContext context, UndertowEmptyPathService delegate) {
-            this.serializers = context.serializerRegistry();
-            this.delegate =
-                    context.serviceInstrumenter()
-                            .instrument(delegate, UndertowEmptyPathService.class);
+        private EmptyPathServiceFactory(
+                UndertowRuntime runtime, UndertowEmptyPathService delegate) {
+            this.runtime = runtime;
+            this.delegate = delegate;
         }
 
-        @Override
-        public void register(EndpointRegistry endpointRegistry) {
-            endpointRegistry.add(
-                    Endpoint.get("/", "EmptyPathService", "emptyPath"), new EmptyPathHandler());
+        List<Endpoint> create() {
+            return ImmutableList.of(new EmptyPathEndpoint());
         }
 
-        private class EmptyPathHandler implements HttpHandler {
+        private class EmptyPathEndpoint implements HttpHandler, Endpoint {
             @Override
             public void handleRequest(HttpServerExchange exchange) throws IOException {
                 boolean result = delegate.emptyPath();
-                serializers.serialize(result, exchange);
+                runtime.serde().serialize(result, exchange);
+            }
+
+            @Override
+            public final HttpString method() {
+                return Methods.GET;
+            }
+
+            @Override
+            public final String template() {
+                return "/";
+            }
+
+            @Override
+            public final Optional<String> serviceName() {
```

Comment:
what are the cases when there's no service name?

Reply:
These are optional so that we may phase out the name in the future if we need to in the future. `serviceName` and `name` are only used internally to record metrics.